src/frontegg-auth/src/auth.rs

```diff
@@ -117,115 +304,222 @@ impl Authentication {
+        // The user is an administrator if they have the admin role that the
+        // `Authenticator` has been configured with.
+        is_admin: msg.claims.roles.iter().any(|r| *r == config.admin_role),
+        _private: (),
+    })
+}
 
-    /// Continuously refreshes and revalidates a [`ValidatedApiTokenResponse`].
-    ///
-    /// The returned future will attempt to refresh the access token before it
-    /// expires, resolving iff the token expires or fails to refresh.
-    ///
-    /// Whenever the access token is successfully refreshed and revalidated,
-    /// `claims_processor` will be invoked with the new validated claims.
-    pub fn continuously_revalidate_api_token_response(
-        &self,
-        mut response: ValidatedApiTokenResponse,
-        mut claims_processor: impl FnMut(&ValidatedClaims),
-    ) -> impl Future<Output = ()> {
-        let frontegg = self.clone();
-
-        // This future resolves once the token expiry time has been reached. It will
-        // repeatedly attempt to refresh the token before it expires.
-        async move {
-            let refresh_url = format!("{}{}", frontegg.admin_api_token_url, REFRESH_SUFFIX);
-            loop {
-                let expire_in = response.claims.exp - frontegg.now.as_secs();
-                let check_in = u64::try_from(expire_in - frontegg.refresh_before_secs).unwrap_or(0);
-                time::sleep(Duration::from_secs(check_in)).await;
-
-                let refresh_request = async {
-                    loop {
-                        let res = async {
-                            let res = frontegg
-                                .client
-                                .refresh_token(&refresh_url, &response.refresh_token)
-                                .await?;
-                            frontegg.validate_api_token_response(res, Some(&response.claims.email))
-                        };
-                        match res.await {
-                            Ok(res) => return res,
-                            Err(e) => {
-                                // Log at info level, not warn or error level,
-                                // because the error could be a result of a
-                                // user's permissions being intentionally
-                                // revoked.
-                                //
-                                // TODO: report Frontegg errors at `error!`
-                                // level or via a Prometheus metric so that
-                                // we can alert on them.
-                                info!(
-                                    "failed to refresh token for {}: {:#}",
-                                    response.claims.email, e
-                                );
-                                // Retry again later. 5 seconds chosen arbitrarily.
-                                time::sleep(Duration::from_secs(5)).await;
```

Comment:
Where did this retry go? Is it still needed?

Reply:
All retry-able errors will already get retried by the underlying client, so this logic was no longer needed.